homeassistant/components/nanoleaf/config_flow.py

```diff
@@ -0,0 +1,272 @@
+            _LOGGER.error("Nanoleaf is not available at %s", self.nanoleaf.host)
+            return self.async_show_form(
+                step_id="user",
+                data_schema=self.user_data_schema,
+                errors={"base": "cannot_connect"},
+                last_step=False,
+            )
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unknown error authorizing Nanoleaf")
+            return self.async_show_form(step_id="link", errors={"base": "unknown"})
+        return await self.async_setup_finish()
+
+    async def async_step_manual(self, user_input: dict[str, Any]) -> FlowResult:
+        """Handle Nanoleaf manual step."""
+        self.nanoleaf.token = user_input[CONF_TOKEN]
+        return await self.async_setup_finish()
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Handle Nanoleaf configuration import."""
+        self._async_abort_entries_match({CONF_HOST: config[CONF_HOST]})
+        _LOGGER.warning(
+            "Importing Nanoleaf on %s from your configuration.yaml", config[CONF_HOST]
+        )
+        self.nanoleaf = Nanoleaf(config[CONF_HOST])
+        self.nanoleaf.token = config[CONF_TOKEN]
+        return await self.async_setup_finish()
+
+    async def async_setup_finish(self) -> FlowResult:
+        """Finish Nanoleaf config flow."""
+        try:
+            info = await self.hass.async_add_executor_job(
+                pynanoleaf_get_info, self.nanoleaf
+            )
+        except Unavailable:
+            _LOGGER.error("Nanoleaf is not available at %s", self.nanoleaf.host)
+            return self.async_show_form(
+                step_id="user",
+                data_schema=self.user_data_schema,
+                errors={"base": "cannot_connect"},
+                last_step=False,
+            )
+        except InvalidToken:
+            _LOGGER.error("Nanoleaf token %s is invalid", self.nanoleaf.token)
+            return self.async_show_form(
+                step_id="manual",
+                data_schema=MANUAL_SCHEMA,
+                errors={"base": "invalid_token"},
+            )
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception(
+                "Unknown error connecting with Nanoleaf at %s with token %s",
+                self.nanoleaf.host,
+                self.nanoleaf.token,
+            )
+            return self.async_show_form(
+                step_id="user",
+                data_schema=self.user_data_schema,
+                errors={"base": "unknown"},
+                last_step=False,
+            )
+        name = info["name"]
+
+        await self.async_set_unique_id(name)
+        self._abort_if_unique_id_configured({CONF_HOST: self.nanoleaf.host})
+
+        # Log successfully imported instructions and optionally update the Nanoleaf config file
+        if self.context["source"] == config_entries.SOURCE_IMPORT:
+            _LOGGER.warning(
```

Comment:
Our log file is not a scratchpad for users. Use persistent_notification

Reply:
I think every integration does this?
A persistent_notification is indeed better, especially because we only show it once and ask for an action from the user.
Changed in https://github.com/home-assistant/core/pull/52199/commits/21bf79ffdc4d4b2adb2a2125737a3948a56d6d24.